flow/envs/green_wave_env.py

```diff
@@ -453,17 +514,19 @@ def k_closest_to_intersection(self, edges, k):
 
         if isinstance(edges, list):
             ids = [self.k_closest_to_intersection(edge, k) for edge in edges]
-            # flatten the list before returning it
+            # flatten the list and return it
             return [veh_id for sublist in ids for veh_id in sublist]
 
         # get the ids of all the vehicles on the edge 'edges' ordered by
-        # increasing distance to intersection
+        # increasing distance to end of edge (intersection)
         veh_ids_ordered = sorted(
```

Comment:
Since we sort the vehicles, it looks like the edge information is lost (which edge the vehicle was on). This appears to be true for the original method as well. While we're at it, should we also return the edge that each vehicle is associated with? And/or we can make it option to sort the vehicles. And/or we can sort the vehicles within each edge, but keep the edges list ordered as it was passed in. This last one, I feel, is the most desirable, at least for the multi-agent settings I'm thinking about.

Reply:
Actually, according to your docstring, the last and most desirable option is indeed what you intended. But the last line in this function `veh_ids_ordered[:k] + pad_lst` seems to add all the padding at the end and not per edge. Am I missing something?

Reply:
Oh. Understood. Recursive call. Awesome.

Reply:
Adding padding solves the information loss, if the parameter is set to True. It might have been more convenient to return a list of list, but then it wouldn't have been backward-compatible and it might be simpler to deal with a flattened list, so I left it like this.